From 2a80dbaf403eeb44b4455e2b93e65407954ab289 Mon Sep 17 00:00:00 2001
From: Ulf Hermann <ulf.hermann@qt.io>
Date: Tue, 13 Jul 2021 11:05:51 +0200
Subject: [PATCH] Directly include Unicode.h in YarrCanonicalize.h

The redirection via utypes.h can be problematic if you have another
utypes.h around in a prominent place. It's easily avoided, though.

Fixes: QTBUG-77528
Change-Id: I50368f56b0d7eb957955900a32dbb625a38d02af
Reviewed-by: Maximilian Goldstein <max.goldstein@qt.io>
Reviewed-by: Fabian Kosmale <fabian.kosmale@qt.io>
Reviewed-by: Eric Lemanissier <eric.lemanissier@gmail.com>
Reviewed-by: Andrei Golubev <andrei.golubev@qt.io>
---
 src/3rdparty/masm/stubs/wtf/unicode/utypes.h | 1 -
 src/3rdparty/masm/yarr/YarrCanonicalize.h    | 2 +-
 2 files changed, 1 insertion(+), 2 deletions(-)
 delete mode 100644 src/3rdparty/masm/stubs/wtf/unicode/utypes.h

diff --git a/src/3rdparty/masm/stubs/wtf/unicode/utypes.h b/src/3rdparty/masm/stubs/wtf/unicode/utypes.h
deleted file mode 100644
index e1b4ff90a6..0000000000
--- a/src/3rdparty/masm/stubs/wtf/unicode/utypes.h
+++ /dev/null
@@ -1 +0,0 @@
-#include <unicode/Unicode.h>
diff --git a/src/3rdparty/masm/yarr/YarrCanonicalize.h b/src/3rdparty/masm/yarr/YarrCanonicalize.h
index cbd279edca..17c9831f19 100644
--- a/src/3rdparty/masm/yarr/YarrCanonicalize.h
+++ b/src/3rdparty/masm/yarr/YarrCanonicalize.h
@@ -26,7 +26,7 @@
 #pragma once
 
 #include <stdint.h>
-#include <unicode/utypes.h>
+#include <unicode/Unicode.h>
 
 namespace JSC { namespace Yarr {
 
